### PR TITLE
Improve PHPCS Exclude list & PHPCS 3rd-party/

### PR DESCRIPTION
**Pull request by kraftbj**
Replaces and closes #10577

#### Changes proposed in this Pull Request:
* PHPCS 3rd-party/ and corresponding tests.
* Add namespacing to a couple of files where there were functions that weren't prefixed with `jetpack`.
* Rename polldaddy.php to crowdsignal.php given the new name of the product.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Tests should still pass.

#### Proposed changelog entry for your changes:
* n/a

### CODE REVIEW

**Comment by kraftbj** [comment]
Not sure what happened to the `excludelist` file; just removed the lines as the files passed.

**Comment by anomiex** [comment]
> Not sure what happened to the `excludelist` file

You changed it from LF to CRLF line endings.

From [some docs](https://docs.github.com/en/free-pro-team@latest/github/using-git/configuring-git-to-handle-line-endings#per-repository-settings) it looks like we can avoid this sort of thing in the future by adding a `.gitattributes` file.

File: 3rd-party/class-domain-mapping.php [hunk 1]
```diff
@@ -7,10 +14,12 @@
  *
  * This class contains methods that are used to provide compatibility between Jetpack sync and domain mapping plugins.
  */
-class Jetpack_3rd_Party_Domain_Mapping {
+class Domain_Mapping {
```
Comment (anomiex) [inline 3rd-party/class-domain-mapping.php:17]: Should the doc comment on line 13 be updated to match the new class name?

File: 3rd-party/class-domain-mapping.php [hunk 2]
```diff
@@ -19,19 +28,27 @@ class Jetpack_3rd_Party_Domain_Mapping {
...
 		if ( is_null( self::$instance ) ) {
-			self::$instance = new Jetpack_3rd_Party_Domain_Mapping;
+			self::$instance = new Domain_Mapping();
 		}
 
 		return self::$instance;
 	}
 
+	/**
+	 * Jetpack_3rd_Party_Domain_Mapping constructor.
```
Comment (anomiex) [inline 3rd-party/class-domain-mapping.php:50]: New class name here too?

File: 3rd-party/vaultpress.php [hunk 3]
```diff
@@ -49,9 +52,10 @@ function jetpack_vaultpress_rewind_enabled_notice() {
...
  */
 function jetpack_vaultpress_rewind_check() {
-	if ( Jetpack::is_active() &&
-		 Jetpack::is_plugin_active( 'vaultpress/vaultpress.php' ) &&
-		 Jetpack::is_rewind_enabled()
+	if (
+		Jetpack::is_active() &&
+		Jetpack::is_plugin_active( 'vaultpress/vaultpress.php' ) &&
+		Jetpack::is_rewind_enabled()
 		) {
```
Comment (anomiex) [inline 3rd-party/vaultpress.php:59]: Feel free to reject, but I find it easier to read like this
```suggestion
	if (
		Jetpack::is_active() &&
		Jetpack::is_plugin_active( 'vaultpress/vaultpress.php' ) &&
		Jetpack::is_rewind_enabled()
	) {
```

File: 3rd-party/woocommerce.php [hunk 4]
```diff
@@ -23,14 +31,14 @@ function jetpack_woocommerce_integration() {
...
-/*
+/**
  * Make sure the social sharing icons show up under the product's short description
  */
 function jetpack_woocommerce_social_share_icons() {
 	if ( function_exists( 'sharing_display' ) ) {
 		remove_filter( 'the_content', 'sharing_display', 19 );
 		remove_filter( 'the_excerpt', 'sharing_display', 19 );
-		echo sharing_display();
+		sharing_display( '' . true );
```
Comment (anomiex) [inline 3rd-party/woocommerce.php:41]: This doesn't seem right. Did you mean
```suggestion
		sharing_display( '', true );
```

File: 3rd-party/woocommerce.php [hunk 5]
```diff
@@ -109,7 +122,8 @@ function jetpack_woocommerce_lazy_images_compat() {
 			}
 			jQuery( 'body' ).get( 0 ).dispatchEvent( jetpackLazyImagesLoadEvent );
 		} );
-	" );
+	"
+	);
```
Comment (anomiex) [inline 3rd-party/woocommerce.php:126]: IMO the close-quote should line up with the open-quote on line 111.
```suggestion
		"
	);
```

File: tests/php/3rd-party/test_class-domain-mapping.php [hunk 6]
```diff
@@ -1,30 +1,78 @@
...
+		parent::tearDown();
+	}
+
+	/**
+	 * Test Setup.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		self::$mock_builder_name = __NAMESPACE__ . '\\MockDomainMapping';
```
Comment (anomiex) [inline tests/php/3rd-party/test_class-domain-mapping.php:61]:
```suggestion
		self::$mock_builder_name = MockDomainMapping::class;
```
Or maybe even better, lose the variable and just use `MockDomainMapping::class` directly where it's needed.

**Review by anomiex — COMMENTED** [review]
One bug [inline](https://github.com/Automattic/jetpack/pull/18061/files/50f77e75c505b5977dbef7c4ca5f5b514ea9d3c9#diff-09fa712bb0443f584fc4974d8c8c812085f57cbcedd493f410622e2aec101e3eR41), and a few suggestions.

File: 3rd-party/class-domain-mapping.php (hunk 1 above)
Comment (kraftbj) [reply 3rd-party/class-domain-mapping.php:17]: 3eaf98765

File: 3rd-party/class-domain-mapping.php (hunk 2 above)
Comment (kraftbj) [reply 3rd-party/class-domain-mapping.php:50]: 3eaf98765

File: tests/php/3rd-party/test_class-domain-mapping.php (hunk 6 above)
Comment (kraftbj) [reply tests/php/3rd-party/test_class-domain-mapping.php:61]: Ah, yes. That's much cleaner as suggested and using directly also makes sense to me. Removes the need for the var and the `setUp` method.

Implemented in d116a942b

**Review by jeherve — APPROVED** [review]

**Review by anomiex — APPROVED** [review]
It's a bit confusing that #18062 seems to have gotten cherry-picked in here rather than being merged in from another branch, so the PR diff is including those changes instead of showing the diff versus current master, but things seem ok as far as I can tell.

**Comment by kraftbj** [comment]
r218953-wpcom

**Pull request merged**